### PR TITLE
Use hfst_optimized_lookup package

### PR DESCRIPTION
**Pull request by andrewdotn**

### CODE REVIEW

**Review by eddieantonio — COMMENTED** [review]
Please merge `master` in again; it should (hopefully) fix the error with the integration tests!

File: CreeDictionary/CreeDictionary/hfstol.py [hunk 1]
```diff
@@ -1,62 +1,21 @@
...
-        output = check_output(
-            [_hfstol, "-q", str(_analyzer_path)],
-            stdin=input_file,
-            stderr=DEVNULL,
-            encoding="UTF-8",
-        )
-
-    raw_analyses = output.split("\n")
-    return parse_analyses(raw_analyses)
+    return parse_analyses(descriptive_analyzer.lookup(wordform))
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/hfstol.py:14]: Ah, I love how simple this is now 🤩

File: CreeDictionary/CreeDictionary/hfstol.py [hunk 2]
```diff
@@ -1,62 +1,21 @@
...
-
-    raw_transductions = output.split("\n")
-    for line in raw_transductions:
-        # Skip empty lines and failures to transduce.
-        if line.strip() == "" or "+?" in line:
-            continue
-
-        _analysis, _tab, wordform = line.partition("\t")
+    for wordform in normative_generator.lookup(analysis):
         yield wordform
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/hfstol.py:19]:
```suggestion
    return normative_generator.lookup(analysis)
```

or even

```suggestion
    yield from normative_generator.lookup(analysis)
```

File: CreeDictionary/utils/paradigm_filler.py [hunk 3]
```diff
@@ -81,7 +82,7 @@ def __init__(self, layout_dir: Path, generator_hfstol_path: Path):
         :param layout_dir: the directory for .layout and .layout.cvs files
         """
         self._layout_tables = self._import_layouts(layout_dir)
-        self._generator = hfstol.HFSTOL.from_file(generator_hfstol_path)
+        self._generator = TransducerFile(generator_hfstol_path)
```
[the remark this answers is not in the document]
Comment (eddieantonio) [reply CreeDictionary/utils/paradigm_filler.py:85]: I will refactor this in a future PR :)

**Review by eddieantonio — APPROVED** [review]
LGTM! This is awesome, and I like how it cleaned up the code quite a lot!

**Review by nienna73 — APPROVED** [review]
These changes look good to me.

**Pull request merged**